tests/test_cookiejar.py

```diff
@@ -272,26 +272,32 @@ async def test_domain_filter_ip_cookie_receive(cookies_to_receive) -> None:
     assert len(jar) == 0
 
 
-async def test_preserving_ip_domain_cookies(loop) -> None:
-    jar = CookieJar(unsafe=True)
-    jar.update_cookies(SimpleCookie(
-        "shared-cookie=first; "
-        "ip-cookie=second; Domain=127.0.0.1;"
-    ))
-    cookies_sent = jar.filter_cookies(URL("http://127.0.0.1/")).output(
-        header='Cookie:')
-    assert cookies_sent == ('Cookie: ip-cookie=second\r\n'
-                            'Cookie: shared-cookie=first')
-
-
-async def test_preserving_quoted_cookies(loop) -> None:
-    jar = CookieJar(unsafe=True)
+@pytest.mark.parametrize(
+    'cookies,expected,quote_bool', [
+        ("shared-cookie=first; ip-cookie=second; Domain=127.0.0.1;",
+         'Cookie: ip-cookie=second\r\nCookie: shared-cookie=first', True),
+        ("ip-cookie=\"second\"; Domain=127.0.0.1;", 'Cookie: ip-cookie=\"second\"', True),
+        ("custom-cookie=value/one;", 'Cookie: custom-cookie="value/one"', True),
+        ("custom-cookie=value1;", 'Cookie: custom-cookie=value1', True),
+        ("custom-cookie=value/one;", 'Cookie: custom-cookie=value/one', False),
+    ]
+)
+async def test_quotes_correctly_based_on_input(loop, cookies, expected, quote_bool) -> None:
+    """
+    Tests the following scenarios:
+     - Preserving IP domain cookies
+     - Preserving quoted cookies
+     - Quoting strings with special chars
+     - Does not quote strings with only alphanumeric chars
+     - Setting quote_cookie=False with special chars skips quotation
```

Comment:
Instead of this, I'd rather set proper param ids in the parameterize declaration above. It allows outputting reasonable test representations in test run reports.